### PR TITLE
Add unior tension tables, allow different left and right spoke size, auto refresh...

### PR DESCRIPTION
**Pull request by plugz**
Hello,

I wanted a tool like this that I could use with the unior spoke tension meter (http://www.unior-bike.com/?doc=10780&prod=758778) and found your simple tool that I could easily update with the unior data !

So first thank you for sharing it.

I've made some changes to it besides adding the unior data:
* Allow having different spoke size on drive side and non drive size: this is done on some rear wheels, including one I have
* Draw a more compact table: having only one spoke per row makes for a too big table for my laptop screen
* Add a "tolerance" setting. When a spoke tension is less that (average - tolerance) or more that (average + tolerance), a ⚠ is displayed right after the tension value
* The entered data is not lost when changing spoke count / spoke type / tolerance
* The external .js (pure, d3, jquery) are now local, so it can work without an internet access
* Number validation is simplified (use of html5 "number" input type)
* Input order is simplified (use of tabindex)
* Also display standard deviation at the bottom of the table

If some changes seem interesting to you, but not all of them, I can split this pull request into more smaller pull requests

### CODE REVIEW

File: wheelchart.js [hunk 1]
```diff
@@ -64,7 +64,8 @@ $(document).ready(function() {
 		for (var i = 0; i < nSpokes; i += 2) {
 			var $row = $('<tr role="valueRow"/>');
 
-			var $input = $('<input type="text" id="reading' + i + '"/>')
+			var tabIndexDS = i / 2 + 1;
```
Comment (KonstantinShemyak) [inline wheelchart.js]: I'm not sure I understand what this commit does. Was there some incorrect behaviour related to the focus, which it fixes?

File: wheelchart.js [hunk 2]
```diff
@@ -129,24 +129,34 @@ $(document).ready(function() {
...
-			$('#' + selector + 'Reading').text(avgReading);
-			$('#' + selector + 'Tension').text(avgTension);
+			var avgReading = sumReading / (nSpokes / 2);
+			var avgTension = sumTension / (nSpokes / 2);
 
 			var variance = Number($varianceInput.val());
 			var minTension = avgTension * ((100 - variance) / 100);
 			var maxTension = avgTension * ((100 + variance) / 100);
 
+			var sqSum = 0;
```
Comment (KonstantinShemyak) [inline wheelchart.js]: Why do we show **standard deviation** here? I would think that the (linear) maximum deviation from the average is the appropriate metric to show? (I'm not saying that showing stddev is wrong - I just do not know the reason why namely stddev, not any other deviation, is appropriate in this case?)

File: wheelchart.js [hunk 3]
```diff
@@ -100,52 +103,63 @@ $(document).ready(function() {
...
-
-			var minTension = avgTension * ((100 - $varianceInput.val()) / 100);
-			var maxTension = avgTension * ((100 + $varianceInput.val()) / 100);
-
-			if (newTension > minTension && newTension < maxTension)
-				$('#tension' + targetSpoke).text(round2(newTension));
-			else
-				$('#tension' + targetSpoke).html(round2(newTension) + " &#9888;");
-
-			/* Move the focus: e.g. for 32-spoke wheel, 31 -> 1, 32 -> 2 */
```
Comment (KonstantinShemyak) [inline wheelchart.js]: I do not want to lose this! :-) The user normally presses "Enter" after typing the number, let's not force them to do more key/mouse actions.

File: wheelchart.js [hunk 4]
```diff
@@ -2,8 +2,7 @@ $(document).ready(function() {
 	'use strict';
 	
 	var config = {
-			INITIAL_READING_DRIVE_SIDE: 20,
-			INITIAL_READING_NON_DRIVE_SIDE: 18,
```
Comment (KonstantinShemyak) [inline wheelchart.js]: I do not want to lose this. The reason is that for a first-time user, showing just one circle is less informative; showing two non-overlapping circles makes it clear (in my opinion) where drive- and non-drive-side spokes are shown, and that they are alternating.

File: index.html [hunk 5]
```diff
@@ -28,6 +28,8 @@ <h1>Wheelbuilding chart</h1>
 				<select id="nSpokes"></select>
 				<span> spokes </span>
 				<select id="spokeThickness"></select>
+				<span> and </span>
```
Comment (KonstantinShemyak) [inline index.html]: The user should see here, which of the dropdowns is for the drive side and which for the non-drive.

File: index.html [hunk 6]
```diff
@@ -33,20 +33,24 @@ <h1>Wheelbuilding chart</h1>
...
+				<span>Variance: </span>
+				<select id="varianceInput"></select>
+			</div>
 			<table id="valuesAndChart">
 				<tr>
 					<td><table id="valuesTable">
 						<tr class="bordered">
-							<td colspan="3" class="driveSideColor header">Drive side spokes</td>
-							<td colspan="3" class="nonDriveSideColor header">Non-drive side spokes</td>
+							<th colspan="3" class="driveSideColor header">Drive side spokes</th>
```
[the remark this answers is not in the document]
Comment (plugz) [reply index.html]: I'll redo everything in separate PRs with cleaner commits

File: wheelchart.js (hunk 1 above)
Comment (plugz) [reply wheelchart.js]: The tabindex attribute allows tabing between spokes in the correct order, without needing the javascript bit that I deleted further down in this file

File: wheelchart.js (hunk 2 above)
Comment (plugz) [reply wheelchart.js]: Honestly, I simply showed the same metric as the park tool app.
It does not add any important value to the tool, I think I'll just remove this line.

File: wheelchart.js (hunk 3 above)
Comment (plugz) [reply wheelchart.js]: Using enter never occured to me, I use tab to switch between inputs.
I'll restore these lines.

**Comment by plugz** [comment]
Hello,

The changes have been split into cleaner commits. It should be easier for you to see what's going on in the changes this way.

I'll also create several pull requests, one for each change.

**Pull request closed without merging**